src/app/public/modules/href/href-resolver.service.ts

```diff
@@ -1,16 +1,16 @@
 import { Injectable } from '@angular/core';
 
 import { SkyHrefResolver } from './href-resolver';
-import { AppRoute } from './types/app-route';
+import { SkyHref } from './types/href';
 
 /**
  * Return the link as-is.
  */
 @Injectable()
 export class SkyHrefResolverService implements SkyHrefResolver {
-  public resolveHref(url: string): Promise<AppRoute> {
-    return Promise.resolve<AppRoute>({
-      url,
+  public resolveHref(param: {url: string}): Promise<SkyHref> {
```

Comment:
Typically when we create one parameter with multiple optional properties we name it `args`. I'd also suggest creating a `SkyHrefResolveArgs` interface and exporting that via the public API for more convenient typing for the consumer, e.g. if the value needs to be assigned to a variable first instead of being passed in as an object literal.